gitlint-core/gitlint/contrib/rules/disallow_cleanup_commits.py

```diff
@@ -0,0 +1,20 @@
+from gitlint.rules import CommitRule, RuleViolation
+
+class DisallowCleanupCommits(CommitRule):
+    """ This rule checks the commits for "fixup!"/"squash!"/"amend!" commits
+    and rejects them.
+    """
+
+    name = "contrib-disallow-cleanup-commits"
+    id = "CC2"
+
+    def validate(self, commit):
+        if commit.is_fixup_commit:
+            return [RuleViolation(self.id, "Fixup commits are not allowed", line_nr=1)]
+
+        if commit.is_squash_commit:
+            return [RuleViolation(self.id, "Squash commits are not allowed", line_nr=1)]
+
+        if commit.message.title.startswith("amend!"):
```

Comment:
I just merged #320, if you rebase to latest `main` branch, you'll be able to do this:
```suggestion
        if commit.is_fixup_amend_commit:
```